File: app/src/main/java/com/simbirsoft_mobile_revolution2017_v_team/mobilelibrary/ui/fragments/FavouriteBooksFragment.java
package com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.ui.fragments;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.R;
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.domain.Book;
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.presenter.LibraryPresenter;
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.view.ILibraryView;
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.ui.fragments.ListBooksFragment.RecyclerViewAdapter;

import java.util.ArrayList;
import java.util.List;

public class FavouriteBooksFragment extends Fragment
        implements RecyclerViewAdapter.OnBookClickListener, ILibraryView {

    private List<Book> library = new ArrayList<>();
    private LibraryPresenter presenter = new LibraryPresenter();
    private RecyclerView mRecyclerView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_favourite_books, container, false);

        //layout'ы лучше именовать under_score rv_favourite
        mRecyclerView = view.findViewById(R.id.rvFavourite);
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));

        presenter.attachView(this);

        return view;
    }

    @Override
    public void onBookClicked(String bookId) {
        // строки все же лучше стараться всегда брать из ресурсов
        Toast.makeText(getActivity().getApplicationContext(),"favourite " + bookId, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onStart() {
        super.onStart();
        presenter.loadFavourites();
    }

    @Override
    public void onDestroy() {
        presenter.detachView();
        super.onDestroy();
    }

    @Override
    public void onDataReceived(List<Book> library) {
        this.library = library;
        mRecyclerView.setAdapter(new RecyclerViewAdapter(this.library, this));
    }

    @Override
    public void onDataReceived(Book book) {

    }

    @Override
    public void onDataCreated(List<Book> library) {

    }

    @Override
    public void onError(Throwable error) {
        Toast.makeText(getActivity().getApplicationContext(),"favourite fail " + String.valueOf(error), Toast.LENGTH_LONG).show();
    }
}

File: app/src/main/java/com/simbirsoft_mobile_revolution2017_v_team/mobilelibrary/ui/fragments/FavouriteBooksFragment.kt
package com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.ui.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.R
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.domain.BookK
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.presenter.LibraryPresenterK
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.ui.fragments.ListBooksFragment.RecyclerViewAdapterK
import com.simbirsoft_mobile_revolution2017_v_team.mobilelibrary.view.ILibraryViewK
import java.util.ArrayList

/**
 * Created by user on 15.11.2017.
 */
class FavouriteBooksFragmentK : Fragment(),
        RecyclerViewAdapterK.OnBookClickListener,
        ILibraryViewK {

    private var library: List<BookK> = ArrayList()
    private val presenter = LibraryPresenterK()
    private var mRecyclerView: RecyclerView? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        val view = inflater!!.inflate(
                R.layout.fragment_favourite_books,
                container,
                false
        )

        mRecyclerView = view.findViewById(R.id.rvFavourite)
        mRecyclerView!!.setHasFixedSize(true)
        mRecyclerView!!.layoutManager = LinearLayoutManager(activity)

        presenter.attachView(this)

        return view
    }

    override fun onBookClicked(bookId: String) {
        Toast.makeText(
                activity.applicationContext,
                "favourite " + bookId,
                Toast.LENGTH_SHORT
        ).show()
    }

    override fun onStart() {
        super.onStart()
        presenter.loadFavourites()
    }

    override fun onDestroy() {
        presenter.detachView()
        super.onDestroy()
    }

    override fun onDataReceived(library: List<BookK>) {
        this.library = library
        mRecyclerView!!.adapter = RecyclerViewAdapterK(this.library, this)
    }

    override fun onDataReceived(book: BookK) {

    }

    override fun onDataCreated(library: List<BookK>) {

    }

    override fun onError(error: Throwable) {
        Toast.makeText(
                activity.applicationContext,
                "favourite fail " + error.toString(),
                Toast.LENGTH_LONG
        ).show()
    }
}